src/napari_spatialdata/_view.py

```diff
@@ -322,6 +292,49 @@ def channel_changed(event: Event) -> None:
         self.model.events.adata.connect(self._on_layer_update)
         self.model.events.color_by.connect(self._change_color_by)
 
+    def _channel_changed(self, event: Event) -> None:
+        layer = self.model.layer
+        is_image = isinstance(layer, Image)
+
+        has_sdata = layer is not None and layer.metadata.get("sdata") is not None
+        has_adata = layer is not None and layer.metadata.get("adata") is not None
+
+        # has_adata is added so we see the channels in the view widget under vars
+        if layer is None or not is_image or not has_sdata or not has_adata:
+            return
+
+        current_point = list(event.value)
+        displayed = self._viewer.dims.displayed
+
+        for i, (lo_size, hi_size, cord) in enumerate(zip(layer.data[-1].shape, layer.data[0].shape, displayed)):
+            if i in displayed:
+                current_point[i] = slice(None)
+            else:
+                current_point[i] = int(cord * lo_size / hi_size)
```

Comment:
This is logic for any dimensional data. 

If you are 100% sure that data will always be 2D, then this logic may be obsolete. 
If contrast limits needs to be calculated for the whole channel, not only the visible portion, then it is also overcomplicated. But then, for 3D data, should be no recalculation when change Z slice.

Reply:
In the future we will not be sure about this. SpatialData already does have 3D models. Where bugs occur for 3D napari-spatialdata should be adjusted.